core/src/main/java/io/syndesis/core/Tokens.java

```diff
@@ -31,9 +33,34 @@
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.UriBuilder;
 import java.io.IOException;
+import java.util.Locale;
+import java.util.function.UnaryOperator;
 
 public final class Tokens {
 
+    public enum TokenProvider implements UnaryOperator<String> {
+        GITHUB {
```

Comment:
@jimmidyson You implemented GITHUB and OPENSHIFT exactly in the opposite wat (openshift has json, github query params)